backend/src/main/java/fk/prof/backend/http/BackendHttpVerticle.java

```diff
@@ -117,13 +136,55 @@ private void handlePostProfile(RoutingContext context) {
         });
   }
 
+  private void handlePostPoll(RoutingContext context) {
+    try {
+      Recorder.PollReq pollReq = ProtoUtil.buildProtoFromBuffer(Recorder.PollReq.parser(), context.getBody());
+      Recorder.ProcessGroup processGroup = RecorderProtoUtil.mapRecorderInfoToProcessGroup(pollReq.getRecorderInfo());
+      ProcessGroupContextForPolling processGroupContextForPolling = this.processGroupDiscoveryContext.getProcessGroupContextForPolling(processGroup);
+      if(processGroupContextForPolling == null) {
+        throw new IllegalArgumentException("Process group " + RecorderProtoUtil.processGroupCompactRepr(processGroup) + " not associated with the backend");
```

Comment:
I see, is it possible to extend IllegalArgumentException and create a BadRequestException and IllegalStateException gets extended to create ServerProblemException (or something else with a better name, that name is pretty obnoxious)? If we can do this, then it'll be possible to throw more sensible looking exceptions, while we still derive benefits from exception-convention.